framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -785,6 +822,18 @@ protected void processCommonAssignment(TransferInput<V, S> in, Node lhs,
         // add new information based on conditional postcondition
         processConditionalPostconditions(n, method, tree, thenStore, elseStore);
 
+        if (inferSignatures) {
+            // Finds the receiver's type
+            Tree receiverTree = n.getTarget().getReceiver().getTree();
+            if (receiverTree == null) {
+                // If there is no receiver, then get the class being visited.
+                receiverTree = analysis.getContainingClass(n.getTree());
```

Comment:
That makes sense, thanks. Please do mention in the comment here that receiverTree will be null in the second assignment in case of a static context.